doc/nativesc/contracts/ForgerStakesV2.md

```diff
@@ -0,0 +1,160 @@
+  
+     Register a new forger.
+     rewardShare can range in [0..1000] and can be 0 if and only if rewardAddress == 0x000..00.<br>
+     Vrf key and signatures are split in two or more separate parameters, being longer than 32 bytes.<br>
+     sign1_x are the 25519 signature chunks and sign2_x are the Vfr signature chunks.<br>
+     The message to sign is a string concatenation of signPubKey+vrfKey+rewardShare+rewardAddress, where rewardAddress
+     is represented in the Eip55 checksum format.<br>
+     The method accepts WEI value: the sent value will be converted to the initial stake assigned to the forger.<br>
+     The initial stake amount must be >= min threshold (10 Zen)
+
+- updateForger
+
+          function updateForger(bytes32 signPubKey, bytes32 vrf1, bytes1 vrf2, uint32 rewardShare, address rewardAddress, bytes32 signature1, bytes32 signature2) external;
+
+
+     Updates an existing forger.<br>
+     A forger can be updated just once and only if rewardAddress == 0x000..00 and rewardShare == 0.<br>
+     Vrf key is split in two separate parameters, being longer than 32 bytes.
+
+- delegate
+
+          function delegate(bytes32 signPubKey, bytes32 vrf1, bytes1 vrf2) external payable;
+  
+     Delegate a stake to a previously registered forger.<br>
+     Vrf key is split in two separate parameters, being longer than 32 bytes.<br>
+     All the ZEN sent with the transaction will be delegated, and the owner will be the sender address.
+
+- withdraw
+
+          function withdraw(bytes32 signPubKey, bytes32 vrf1, bytes1 vrf2, uint256 amount) external;
+  
+     Withdraw (unstake) a previously assigned stake.<br>
+     Vrf key is split in two separate parameters, being longer than 32 bytes.
+
+- stakeTotal
+
+          function stakeTotal(bytes32 signPubKey, bytes32 vrf1, bytes1 vrf2, address delegator, uint32 consensusEpochStart, uint32 maxNumOfEpoch) external view returns (uint256[] memory listOfStakes);
+
+
+     Returns the total stake amount, at the end of one or more consensus epochs, assigned to a specific forger.<br>
+     vrf, signKey and delegator are optional: if all are null, the total stake amount will be returned. If only
+     delegator is null, all the stakes assigned to the forger will be summed.
+     If vrf and signKey are null, but delegator is defined, the method will fail.<br>
+     consensusEpochStart and maxNumOfEpoch are optional: if both null, the data at the current consensus epoch is returned.
+     Returned array contains also elements with 0 value. Returned values are ordered by epoch, and the array length may
+     be < maxNumOfEpoch if the current consensus epoch is < (consensusEpochStart + maxNumOfEpoch) or if the forger was
+     registered after consensusEpochStart.
+
+- rewardsReceived
+
+          function rewardsReceived(bytes32 signPubKey, bytes32 vrf1, bytes1 vrf2, uint32 consensusEpochStart, uint32 maxNumOfEpoch) external view returns (uint256[] memory listOfRewards);
```

Comment:
same as stakeTotal with the only difference that the rewards corresponding to the current epoch are never returned